src/core/scadstream.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <ostream>
+#include <sstream>
+#include <string>
+#include <iterator>
+#include <double-conversion/double-conversion.h>
+
```

Comment:
IMHO would be preferable to make these defines static members of the scad::ostringstream class 
```c++
class ostringstream
{
public:
   using dc_flags_t = double_conversion::DoubleToStringConverter::Flags;
   static constexpr dc_flags_t dc_flags = dc_flags_t::UNIQUE_ZERO;
   static constexpr std::intmax_t dc_buffer_size = 128;
   static constexpr char dc_inf[] = "inf";
   static constexpr char dc_nan[] = "nan";
   static constexpr char dc_exp = 'e';
   static constexpr int dc_decimal_low_exp = -5;
   static constexpr int dc_decimal_high_exp = 6;
   static constexpr int dc_max_leading_zeros = 5;
   static constexpr int dc_max_trailing_zeros = 0;
  ostringstream() :
      builder(buffer, dc_buffer_size),
      dc(dc_flags, dc_inf, dc_nan, dc_exp,
          dc_decimal_low_exp, dc_decimal_high_exp,
          dc_max_leading_zeros, dc_max_trailing_zeros)
  {}
...
};
```

Reply:
Good point, always nice to replace `#define`s.  Those were moved over from Value.cc without much thought to refactoring.